services/rpcfilters/transaction_sent_to_upstream_event.go

```diff
@@ -0,0 +1,98 @@
+package rpcfilters
+
+import (
+	"errors"
+	"sync"
+
+	"github.com/ethereum/go-ethereum/common"
+)
+
+// transactionSentToUpstreamEvent represents an event that one can subscribe to
+type transactionSentToUpstreamEvent struct {
+	sxMu     sync.Mutex
+	sx       map[int]chan common.Hash
+	listener chan bool
+	quit     chan struct{}
+}
+
+func (e *transactionSentToUpstreamEvent) Start() error {
+	if e.quit != nil {
+		return errors.New("latest transaction sent to upstream event is already started")
+	}
+
+	e.quit = make(chan struct{})
+
+	go func() {
+		for {
+			select {
+			case <-e.listener:
+				if e.numberOfSubscriptions() == 0 {
+					continue
+				}
+				e.processTransactionSentToUpstream()
+			case <-e.quit:
+				return
+			}
+		}
+	}()
+
+	return nil
+}
+
+func (e *transactionSentToUpstreamEvent) numberOfSubscriptions() int {
+	e.sxMu.Lock()
+	defer e.sxMu.Unlock()
+	return len(e.sx)
+}
+
+func (e *transactionSentToUpstreamEvent) processTransactionSentToUpstream() {
+
+	e.sxMu.Lock()
+	defer e.sxMu.Unlock()
+
+	for _, channel := range e.sx {
+		// Send in an empty hash for now.
+		channel <- common.Hash{}
```

Comment:
@mandrigin, @adambabik Here I am simply sending in an empty hash. In the case of the `latestBlockChangedEvent`, this hash is related to the block change itself, so I'm wondering if the `rpcfilters.filter` (in api.go) struct uses this hash outside the context of the block change event (like if it was used to simply track all the different times the event has been triggered). If I still do want to send the hash into the channel, what would be a good value for it?

Reply:
Shouldn't it be a pending transaction hash? At least that's the conclusion after following [the implementation in go-ethereum](https://github.com/ethereum/go-ethereum/blob/953b5ac015da7fdef39c6864bb0cb490ffaa7959/eth/filters/api.go#L98-L134).